### PR TITLE
Add .NET 5.0 support

### PR DESCRIPTION
**Pull request by pepone**
This PR add support for .NET 5.0, the C# NuGet packages now include additional assemblies in lib/net5.0 that target .NET 5.0.
msbuild\ice.5.0.sln and msbuild\ice.5.0.test.sln solution files have been added to build Ice with .NET 5.0.

The .NET 5.0 build can be tested with `allTest.py --framework=net5.0`

### CODE REVIEW

File: csharp/README.md [hunk 1]
```diff
@@ -19,7 +19,8 @@ resulting binaries. As an alternative, you can download and install the
 
 ## Building on Windows
 
-A source build of Ice for .NET on Windows produces two sets of assemblies:
+A source build of Ice for .NET on Windows produces three sets of assemblies:
+ - assemblies for [.NET 5.0][8]
```
Comment (bernardnormier) [inline csharp/README.md:23]: We should resort the references.

File: csharp/README.md [hunk 2]
```diff
@@ -45,23 +46,26 @@ To build all Ice assemblies and the associated test suite, run:
...
 
 You can skip the build of the test suite with the `BuildDist` target:
 ```
 msbuild msbuild\ice.proj /t:BuildDist
 ```
 
-The `Net45Build`, `Net45BuildDist`, `NetStandardBuild` and
-`NetStandardBuildDist` targets allow you to build assemblies only for the .NET
+The `Net5Build`, `Net5BuildDist`, `Net45Build`, `Net45BuildDist`, `NetStandardBuild` and
+`NetStandardBuildDist` targets allow you to build assemblies only for .NET 5, the .NET
```
Comment (bernardnormier) [inline csharp/README.md:64]: Should be .NET 5.0

File: csharp/README.md [hunk 3]
```diff
@@ -116,8 +120,8 @@ necessary.
 
 ### Linux and macOS Build Requirements
 
-You need the [.NET Core 2.1 SDK][4] or [.NET Core 3.1 SDK][5] to build
-Ice for .NET from source.
+You need the [.NET Core 2.1 SDK][4], [.NET Core 3.1 SDK][5] or [.NET 5 SDK][6]
```
Comment (bernardnormier) [inline csharp/README.md:123]: .NET 5.0

File: csharp/src/IceBox/msbuild/icebox/net5.0/icebox.csproj [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  <PropertyGroup>
+    <UseAppHost>true</UseAppHost>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="../../../AssemblyInfoExe.cs" />
+    <Compile Include="../../../Server.cs" />
+    <Compile Include="../../../ServiceManagerI.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="../../../../Ice//msbuild/net5.0//ice.csproj" />
```
Comment (bernardnormier) [inline csharp/src/IceBox/msbuild/icebox/net5.0/icebox.csproj:20]: Why all these `//`?

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: csharp/src/IceBox/msbuild/icebox/net5.0/icebox.csproj (hunk 4 above)
Comment (pepone) [reply csharp/src/IceBox/msbuild/icebox/net5.0/icebox.csproj:20]: probably an error with the sed command I use to rewrite the references, I will fix it

File: csharp/.gitignore [hunk 5]
```diff
@@ -11,3 +11,9 @@ IceSSL.XML
 IceStorm.XML
 test/**/netstandard2.0/netcoreapp2.0
 test/**/netstandard2.0/net4*
+test/**/net5.0/client
+test/**/net5.0/collocated
+test/**/net5.0/serveramd
+test/**/net5.0/serveramdtie
+test/**/net5.0/servertie
+test/**/net5.0/server
```
Comment (externl) [inline csharp/.gitignore:19]: Not a big deal, but there's a missing newline here

**Review by bentoi — APPROVED** [review]

File: csharp/.gitignore (hunk 5 above)
Comment (pepone) [reply csharp/.gitignore]: Fixed

File: csharp/src/IceBox/msbuild/icebox/net5.0/icebox.csproj (hunk 4 above)
Comment (pepone) [reply csharp/src/IceBox/msbuild/icebox/net5.0/icebox.csproj]: Fixed

**Pull request merged**